Content.Server/Weapons/Ranged/Systems/GunSystem.cs

```diff
@@ -313,6 +315,26 @@ private void ShootOrThrow(EntityUid uid, Vector2 mapDirection, Vector2 gunVeloci
         ShootProjectile(uid, mapDirection, gunVelocity, gunUid, user, gun.ProjectileSpeed);
     }
 
+
+    public void ShootProjectile(EntityUid uid, Vector2 direction, Vector2 gunVelocity, EntityUid gunUid, EntityUid? user = null, float speed = 20f)
+    {
+        var physics = EnsureComp<PhysicsComponent>(uid);
+        Physics.SetBodyStatus(physics, BodyStatus.InAir);
+
+        var targetMapVelocity = gunVelocity + direction.Normalized() * speed;
+        var currentMapVelocity = Physics.GetMapLinearVelocity(uid, physics);
+        var finalLinear = physics.LinearVelocity + targetMapVelocity - currentMapVelocity;
+        Physics.SetLinearVelocity(uid, finalLinear, body: physics);
+
+        var projectile = EnsureComp<ProjectileComponent>(uid);
+        //in the situation when user == null, means that the cannon fires on its own (via signals). And we need the gun to not fire by itself in this case
+        Projectiles.SetShooter(uid, projectile, user ?? uid);
+        projectile.Weapon = gunUid;
+
+
+        TransformSystem.SetWorldRotation(uid, direction.ToWorldAngle());
+    }
+    
```

Comment:
This hides the shared method which you shouldn't do as callers might not call the right one and throws up an IDE warning.

Reply:
Wat, is it no my function

Reply:
I change only 331 stroke